Sources/TSCBasic/Process/ProcessEnv.swift

```diff
@@ -11,23 +11,77 @@
 import Foundation
 import TSCLibc
 
+public struct CaseInsensitiveString {
+  public let value: String
+  public init(_ value: String) {
+    self.value = value
+  }
+}
+
+extension CaseInsensitiveString: Equatable {
+  public static func == (_ lhs: Self, _ rhs: Self) -> Bool {
+    // TODO: is this any faster than than just doing a lowercased conversion and compare?
+    return lhs.value.caseInsensitiveCompare(rhs.value) == .orderedSame
+  }
+}
+
+extension CaseInsensitiveString: ExpressibleByStringLiteral {
+  public init(stringLiteral value: String) {
+    self.init(value)
+  }
+}
+
+extension CaseInsensitiveString: Hashable {
+  public func hash(into hasher: inout Hasher) {
+    self.value.lowercased().hash(into: &hasher)
+  }
+}
+
+#if os(Windows)
+public typealias ProcessEnvironmentBlock = [CaseInsensitiveString:String]
+extension ProcessEnvironmentBlock {
+  public init(_ dictionary: [String:String]) {
+    self.init(uniqueKeysWithValues: dictionary.map { (CaseInsensitiveString($0.key), $0.value) })
+  }
+}
+#else
+public typealias ProcessEnvironmentBlock = [String:String]
```

Comment:
I'm unsure if this is a good name, since AFAIU this is specific to Windows NT. Is there something more generic we could use instead?

Reply:
We already have a mixture of `env` and `environment`, I think that this is a reasonable alternative.